src/model_manager/README.md

```diff
@@ -0,0 +1,216 @@
+# Model Manager
+
+A Python module for automated ML model download, verification, and caching from HuggingFace.
+
+## Features
+
+- **Automated Download**: Download models from HuggingFace Hub with support for specific revisions and file filtering
+- **Integrity Verification**: Verify model integrity using size or SHA256 checksums
+- **Smart Caching**: Skip downloads for already-cached models
+- **CI-Friendly**: Environment variable-based config selection for minimal CI downloads
+- **CLI & Programmatic API**: Use from command line or import as a Python module
+
+## Quick Start
+
+### CLI Usage
+
+```bash
+# Download all models (uses config/model_manager/models.yaml by default)
+python -m model_manager
+
+# Use a specific config file
+python -m model_manager --config config/model_manager/models.yaml
+
+# Download a specific model only
+python -m model_manager --model category_classifier_modernbert-base_model
+
+# List all configured models and their cache status
+python -m model_manager --list
+
+# Verify existing models without downloading
+python -m model_manager --verify-only
+
+# Clean all cached models
+python -m model_manager --clean
+
```

Comment:
You can clean specific models using
`python -m model_manager --config config/models.yaml --clean`.